Migrate nginx directives reference (#1620)
Kong/developer.konghq.com: app/gateway/nginx-directives.md

that file, meaning that the custom server defined in it will be responding,
alongside the regular {{site.base_gateway}} ports:

```bash
curl -I http://127.0.0.1:2112
HTTP/1.1 200 OK
...
```

If you use a relative path in an `nginx_http_include` property, that
path will be interpreted relative to the value of the `prefix` property of
your `kong.conf` file (or the value of the `-p` flag of `kong start` if you
used it to override the prefix when starting {{site.base_gateway}}).

## Custom Nginx templates and embedding {{site.base_gateway}}

You can use custom Nginx
configuration templates directly in two cases: 

- You must modify {{site.base_gateway}}'s default
Nginx configuration. Specifically values that are not adjustable in `kong.conf`, you can modify the template used by {{site.base_gateway}} for producing its
Nginx configuration and launch {{site.base_gateway}} using your customized template.

- You must embed {{site.base_gateway}} in an already running OpenResty instance, you
can reuse {{site.base_gateway}}'s generated configuration and include it in your existing
configuration.

### Custom Nginx templates

> {{site.base_gateway}} can be started, reloaded and restarted with an `--nginx-conf` argument,

Author: closing this comment because I used the other suggestion for the same line

which must specify an Nginx configuration template. The template uses the
[Penlight](http://stevedonovan.github.io/Penlight/api/index.html) [templating engine](http://stevedonovan.github.io/Penlight/api/libraries/pl.template.html), which is compiled using
the {{site.base_gateway}} configuration.

The following Lua functions are available in the [templating engine](http://stevedonovan.github.io/Penlight/api/libraries/pl.template.html):

- `pairs`, `ipairs`
- `tostring`
- `os.getenv`

> The default template for
> {{site.base_gateway}} can be found using this command on the system
> running your {{site.base_gateway}} instance:
> `find / -type d -name "templates" | grep kong`.
> For open-source {{site.base_gateway}}, you can also see the
> [templates directory](https://github.com/kong/kong/tree/master/kong/templates).

Reviewer: ````suggestion
You can find the default template by using this command on the system running your {{site.base_gateway}} instance:
```
find / -type d -name "templates" | grep kong
```
````

> The template is split in two
> Nginx configuration files: `nginx.lua` and `nginx_kong.lua`. The former is
> minimal and includes the latter, which contains everything {{site.base_gateway}} requires
> to run. When `kong start` runs, right before starting Nginx, it copies these
> two files into the prefix directory, which looks like so:
> 

Reviewer: ```suggestion
The template is split in two
Nginx configuration files: `nginx.lua` and `nginx_kong.lua`. 
`nginx.lua` is minimal and includes `nginx_kong.lua`, which contains everything {{site.base_gateway}} requires to run. 
When `kong start` runs, it copies both files into the prefix directory right before starting Nginx, which looks like this:
```

```
/usr/local/kong
├── nginx-kong.conf
└── nginx.conf
```

> If you must tweak global settings that are defined by {{site.base_gateway}} but not adjustable
> via the {{site.base_gateway}} configuration in `kong.conf`, you can inline the contents of the
> `nginx_kong.lua` configuration template into a custom template file (in this
> example called `custom_nginx.template`) like this:

Reviewer: ```suggestion
If you need to adjust global settings that are defined by {{site.base_gateway}} but aren't configurable via parameters in `kong.conf`, you can inline the contents of the
`nginx_kong.lua` configuration template into a custom template file. 
For example, the following file named `custom_nginx.template` adjusts some logging settings:
```
